homeassistant/components/light/homekit_controller.py

```diff
@@ -0,0 +1,182 @@
+"""
+Support for Homekit lights.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.homekit/
+"""
+import colorsys
+import json
+import logging
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_RGB_COLOR, ATTR_COLOR_TEMP, SUPPORT_BRIGHTNESS,
+    SUPPORT_RGB_COLOR, SUPPORT_COLOR_TEMP, Light)
+
+REQUIREMENTS = ['homekit==0.5']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, service):
+    """Set up Homekit lighting."""
+    add_devices([HomekitLight(service)])
+
+
+class HomekitLight(Light):
+    """Representation of a Homekit light."""
+
+    def __init__(self, service):
+        """Initialize the light."""
+
+        self._name = service['device'].model
+        self._securecon = service['device'].securecon
+        self._aid = service['aid']
+        self._iid = service['service']['iid']
+        self._address = "%s-%s" % (service['device'].hkid, service['aid'])
+        self._features = 0
+        self._chars = {}
+        self.update_characteristics(service['service']['characteristics'])
+
+    def update_state(self):
+        response = self._securecon.get('/accessories')
+        data = json.loads(response.read().decode())
+        for accessory in data['accessories']:
```

Comment:
This seems very inefficient, no? Each entity will fetch the state for all accessories?